code/_globalvars/logging.dm

```diff
@@ -1,55 +1,73 @@
 GLOBAL_LIST_EMPTY(active_turfs_startlist)
 
-GLOBAL_LIST_EMPTY(admin_log)
-GLOBAL_PROTECT(admin_log)
-
-GLOBAL_LIST_EMPTY(adminlog)
-GLOBAL_PROTECT(adminlog)
-
-GLOBAL_LIST_EMPTY(bombers)
-GLOBAL_PROTECT(bombers)
-
-GLOBAL_LIST_EMPTY(combatlog)
-GLOBAL_PROTECT(combatlog)
-
-GLOBAL_VAR(config_error_log)
-GLOBAL_PROTECT(config_error_log)
-
-GLOBAL_VAR(demo_log)
-GLOBAL_PROTECT(demo_log)
-
-GLOBAL_VAR(dynamic_log)
-GLOBAL_PROTECT(dynamic_log)
-
-GLOBAL_VAR(filter_log)
-GLOBAL_PROTECT(filter_log)
-
-#ifdef REFERENCE_DOING_IT_LIVE
-GLOBAL_LIST_EMPTY(harddel_log)
-GLOBAL_PROTECT(harddel_log)
-#endif
-
-GLOBAL_LIST_EMPTY(IClog)
-GLOBAL_PROTECT(IClog)
-
-/// Keeps last 100 signals here in format: "[src] used [REF(src)] @ location [src.loc]: [freq]/[code]"
-GLOBAL_LIST_EMPTY(lastsignalers)
-GLOBAL_PROTECT(lastsignalers)
+GLOBAL_VAR(round_id)
+GLOBAL_PROTECT(round_id)
 
-/// Stores who uploaded laws to which silicon-based lifeform, and what the law was
-GLOBAL_LIST_EMPTY(lawchanges)
-GLOBAL_PROTECT(lawchanges)
+/// The directory in which ALL log files should be stored
 GLOBAL_VAR(log_directory)
 GLOBAL_PROTECT(log_directory)
 
-GLOBAL_VAR(lua_log)
-GLOBAL_PROTECT(lua_log)
+#define DECLARE_LOG_NAMED(log_var_name, log_file_name, start)\
+GLOBAL_VAR(##log_var_name);\
+GLOBAL_PROTECT(##log_var_name);\
+/world/_initialize_log_files(temp_log_override = null){\
+	..();\
+	GLOB.##log_var_name = temp_log_override || "[GLOB.log_directory]/[##log_file_name].log";\
+	if(!temp_log_override && ##start){\
+		start_log(GLOB.##log_var_name);\
+	}\
+}
+
+#define DECLARE_LOG(log_name, start) DECLARE_LOG_NAMED(##log_name, "[copytext(#log_name, 1, length(#log_name) - 4)]", start)
+
+/// Populated by log declaration macros to set log file names and start messages
+/world/proc/_initialize_log_files(temp_log_override = null)
+	// Needs to be here to avoid compiler warnings
+	SHOULD_CALL_PARENT(TRUE)
+	return
+
+// All individual log files
+DECLARE_LOG(config_error_log, FALSE)
+DECLARE_LOG(dynamic_log, FALSE)
+DECLARE_LOG(lua_log, FALSE)
+DECLARE_LOG(perf_log, FALSE) // Declared here but name is set in time_track subsystem
+DECLARE_LOG(query_debug_log, FALSE)
+DECLARE_LOG(signals_log, FALSE)
+DECLARE_LOG(tgui_log, TRUE)
+#ifdef REFERENCE_DOING_IT_LIVE
+DECLARE_LOG_NAMED(harddel_log, "harddels", TRUE)
+#endif
+#if defined(UNIT_TESTS) || defined(SPACEMAN_DMM)
+DECLARE_LOG_NAMED(test_log, "tests", TRUE)
+#endif
+DECLARE_LOG_NAMED(filter_log, "filters", FALSE)
```

Comment:
These TRUE and FALSE are indecipherable--make two separate macros for whatever you're doing here

Reply:
It's whether or not it calls `start_log`. Adding another macro would require another copy of the macro function above. You sure?

Reply:
just make two defines, DO_START_LOG, DONT_START_LOG

Reply:
Good suggestion